### PR TITLE
Feature/grid order prop

### PR DESCRIPTION
**Pull request by brunottonurb**
Add non-responsive `order` prop to reorder `Columns` in `Grid`. Responsive usage is similar to `sizeOf ` prop, e.g. `small-order="3"` or `medium-order="4"`.

### CODE REVIEW

File: stories/Grid.js [hunk 1]
```diff
@@ -102,6 +102,46 @@ stories
...
+        sizeOf="1-2"
+        order="1"
+        small-order="2"
+        medium-order="3"
+      >
+        <Box>One. Try adjusting the window width!</Box>
+      </Column>
+      <Column
+        className="slds-m-bottom--large slds-p-horizontal_small"
+        sizeOf="1-2"
```
Comment (fspoettel) [inline stories/Grid.js]: You could go with a `medium-sizeOf` here, the demo looks broken when resizing

File: src/components/Grid/Column.js [hunk 2]
```diff
@@ -13,8 +13,9 @@ const validBreakpoints = [
   'max-large',
 ];
 
-const breakPointProp = breakpoint => `${breakpoint}-sizeOf`;
+const breakPointProp = (breakpoint, suffix) => `${breakpoint}-${suffix}`;
 const sizeRegex = /^([1-9]|1[0-2])-([1-9]|1[0-2])$/;
+const orderRegex = /^((1[0,1,2])|[1-9])$/;
```
Comment (fspoettel) [inline src/components/Grid/Column.js]: the regex is rather simple in what it does, maybe a functional condition with Math.min/Math.max would be better here? You could keep the regex for consistency's sake, but the function will be more performant.

File: src/components/Grid/Column.js [hunk 3]
```diff
@@ -36,18 +37,27 @@ const Column = (props) => {
...
+    const orderString = breakpoint === '' ? 'order' : `${breakpoint}-order`;
+    const order = props[orderPropName];
+    if (orderRegex.test(order)) {
+      const position = parseInt(order.split('-')[0], 10);
+      sldsClasses.push(`slds-${orderString}_${position}`);
+    }
   });
 
-  const restProps = omit(rest, validBreakpoints.map(breakPointProp), 'sizeOf');
+  let restProps = rest;
```
Comment (fspoettel) [inline src/components/Grid/Column.js]: You could remove one Iteration and the triple assignment by changing to:

``` js
const generatedProps = [];

validBreakpoints.forEach((a) => {
  generatedProps.push(
    breakPointProp(a, 'sizeOf')), 'sizeOf'),
    breakPointProp(a, 'order')), 'order'),
  );
});

const restProps = omit(rest, generatedProps)
```

File: src/components/Grid/Column.js [hunk 4]
```diff
@@ -36,18 +37,27 @@ const Column = (props) => {
...
-
-    const size = props[propName];
+    const size = props[sizePropName];
     if (sizeRegex.test(size)) {
       const from = parseInt(size.split('-')[0], 10);
       const to = parseInt(size.split('-')[1], 10);
       sldsClasses.push(`slds-${sizeString}_${from}-of-${to}`);
     }
+
+    const orderPropName = breakpoint === '' ? 'order' : breakPointProp(breakpoint, 'order');
```
Comment (fspoettel) [inline src/components/Grid/Column.js]: `breakpoint === ''` can be stored in a const here.

File: src/components/Grid/Column.js (hunk 2 above)
Comment (brunottonurb) [reply src/components/Grid/Column.js]: The prop can be a string as well as a number. I could use parseInt and isNaN to check the prop. Are regex really that slow?

Comment (fspoettel) [reply src/components/Grid/Column.js]: We can keep it here, but in general, JS regexes are slower by several orders of magnitude than other operations

**Pull request merged**